src/Application/Interfaces/IGenericRepository.cs

```diff
@@ -1,16 +1,21 @@
-﻿using System;
+﻿using Domain.Common;
+using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Linq.Expressions;
 using System.Text;
 using System.Threading.Tasks;
 
 namespace Application.Interfaces
 {
-    public interface IGenericRepository<T> where T : class
+    public interface IGenericRepository<T>: IDisposable where T : class
     {
-        T GetById(int id);
-        IEnumerable<T> GetAll();        
-        void Add(T entity);        
-        void Remove(T entity);        
+        Task Add(T entity);
+        Task<List<T>> GetAll();
+        Task<T> GetById(Guid id);
```

Comment:
please add  async for all async methods -> AddAsync() ..etc

Reply:
added async in implememntation